site/app/controllers/forum/ForumController.php

```diff
@@ -916,6 +916,22 @@ public function getThreads($page_number = null) {
      * @Route("/courses/{_semester}/{_course}/forum/threads/single", methods={"POST"})
      */
     public function getSingleThread() {
+        // Checks if thread id is undefined. If so, render "fail" json response case informing that thread id is not set.
+        if (!isset($_POST['thread_id'])) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (UNSET ID)");
+        }
+        // Checks if thread id is null. If so, render "fail" json response case informing that thread id is null.
+        if ($_POST['thread_id'] == null) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NULL ID)");
+        }
+        // Checks if thread id is not an integer value. If so, render "fail" json response case informing that thread id is not an integer value.
+        if (!(is_int($_POST['thread_id']) || ctype_digit($_POST['thread_id']))) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NON-INTEGER)");
+        }
+        // Checks if thread id does not exist. If so, render "fail" json response case informing that the thread does not exist.
+        if (!$this->core->getQueries()->existsThread($_POST['thread_id'])) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NON-EXISTENT ID)");
+        }
```

Comment:
You could probably just check the result of the below getThread function call rather than doing this.

Reply:
Okay.  I'll use `!(count($thread) > 1)` instead of `!$this->core->getQueries()->existsThread($thread_id)`.  Thank you.